File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int currentSum = 0;
        for(int num: nums){
            currentSum += num;
            if(currentSum > max){
                max = currentSum;
            }
            if(currentSum < 0 ){
                currentSum = 0;
            }
        }
        return max;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let max = -Infinity;
    let currentSum = 0;
    for(let num of nums){
        currentSum += num;
        if(currentSum > max){
            max = currentSum;
        }
        if(currentSum < 0){
            currentSum = 0;
        }
    }
    return max;
};